src/Core/Checkout/Cart/Order/OrderConverter.php

```diff
@@ -391,10 +404,21 @@ public function assembleSalesChannelContext(OrderEntity $order, Context $context
         return $salesChannelContext;
     }
 
-    private function convertDeliveries(OrderDeliveryCollection $orderDeliveries, LineItemCollection $lineItems): DeliveryCollection
+    private function convertDeliveries(?string $primaryOrderDeliveryId, OrderDeliveryCollection $orderDeliveries, LineItemCollection $lineItems): DeliveryCollection
     {
+        // Ensure primary delivery is first, so `$deliveries->first()` returns the primary delivery.
+        $keys = array_unique([$primaryOrderDeliveryId, ...$orderDeliveries->getKeys()]);
+
+        if ($primaryOrderDeliveryId === null) {
+            $keys = \array_slice($keys, 1);
+        }
```

Comment:
Do not sort before 6.8
```suggestion
        $keys = \array_filter(\array_unique([$primaryOrderDeliveryId, ...$orderDeliveries->getKeys()]));

        if (!Feature::isActive('v6.8.0.0')) {
            $keys = $orderDeliveries->getKeys();
        }
```